### PR TITLE
include keycloak-enabled tests in travis build

### PR DESCRIPTION
**Pull request by ryandawsonuk**
For #1245
Have moved keycloak-enabled tests out of activiti-services-identity-keycloak as they are integration tests that need a full application running. They are now in the starter. This is ok as starter dependencies can be overridden with maven if alternative IDM is used. Have accordingly updated the starter tests to be keycloak-enabled. Also updated the README files to fit changes to project structure.

### CODE REVIEW

File: activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter/pom.xml [hunk 1]
```diff
@@ -131,5 +136,25 @@
...
+        <artifactId>maven-surefire-plugin</artifactId>
+        <version>${maven-surefire-plugin.version}</version>
+        <configuration>
+          <redirectTestOutputToFile>true</redirectTestOutputToFile>
+          <failIfNoTests>false</failIfNoTests>
+          <trimStackTrace>false</trimStackTrace>
+          <excludes>
+            <exclude>**/*TestCase.java</exclude>
+            <exclude>**/*IT.java</exclude>
+            <!--no longer exclude keycloak as travis should be running it
```
Comment (Salaboy) [inline activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter/pom.xml]: @ryandawsonuk what's the point of having the comments here?

File: activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter/src/test/java/org/activiti/cmdendpoint/CommandEndpointKeycloakIT.java [hunk 2]
```diff
@@ -48,9 +50,9 @@
 @RunWith(SpringRunner.class)
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 @TestPropertySource("classpath:application-test.properties")
-@DirtiesContext
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_CLASS)
 @EnableBinding(MessageClientStream.class)
-public class CommandEndpointTest {
+public class CommandEndpointKeycloakIT extends KeycloakEnabledBaseTestIT {
```
Comment (Salaboy) [inline activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter/src/test/java/org/activiti/cmdendpoint/CommandEndpointKeycloakIT.java]: @ryandawsonuk I;m worried about having all the tests called KeycloakIT now... there is no need if we assume that keycloak will be there always right?

File: activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter/src/test/java/org/activiti/definition/ProcessDefinitionKeycloakIT.java [hunk 3]
```diff
@@ -35,7 +37,8 @@
 
 @RunWith(SpringRunner.class)
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
-public class ProcessDefinitionIT {
+@TestPropertySource("classpath:application-test.properties")
+public class ProcessDefinitionKeycloakIT extends KeycloakEnabledBaseTestIT{
```
Comment (Salaboy) [inline activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter/src/test/java/org/activiti/definition/ProcessDefinitionKeycloakIT.java]: same here

File: activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter/src/test/java/org/activiti/keycloak/KeycloakEnabledBaseTestIT.java [hunk 4]
```diff
@@ -55,11 +55,11 @@ protected AccessTokenResponse authenticateUser() throws IOException {
 
     protected HttpHeaders getHeaders(String token) {
         HttpHeaders headers = new HttpHeaders();
-        headers.add("Authorization", "Bearer " + new String(token));
+        headers.add("Authorization", "Bearer " + token);
```
Comment (Salaboy) [inline activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter/src/test/java/org/activiti/keycloak/KeycloakEnabledBaseTestIT.java]: @ryandawsonuk just curious... is this a normal practice... I would assume that spring will inject this for us.

File: pom.xml [hunk 5]
```diff
@@ -631,7 +631,8 @@
           <excludes>
             <exclude>**/*TestCase.java</exclude>
             <exclude>**/*IT.java</exclude>
-            <exclude>**/*Keycloak*.java</exclude>
+            <!--no longer exclude keycloak as travis should be running it
```
Comment (Salaboy) [inline pom.xml]: @ryandawsonuk same.. remove the comments.. they are dead code that we don't want to maintain

File: pom.xml [hunk 6]
```diff
@@ -869,11 +870,12 @@
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-failsafe-plugin</artifactId>
           <version>${maven-failsafe-plugin.version}</version>
+          <!--no longer exclude keycloak as travis should be running it
```
Comment (Salaboy) [inline pom.xml]: @ryandawsonuk same here

File: activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter/src/test/java/org/activiti/keycloak/KeycloakEnabledBaseTestIT.java (hunk 4 above)
Comment (ryandawsonuk) [inline activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter/src/test/java/org/activiti/keycloak/KeycloakEnabledBaseTestIT.java]: If it were always the same token then I could add it to the template but when we pass keycloak auth we then get a temporary token which we use for the life of that particular test. In theory if we had a really long-running test we'd have to call authenticateUser() again and get a new token (which would be simple to do as the methods are there in the base test).

Comment (ryandawsonuk) [inline activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter/src/test/java/org/activiti/keycloak/KeycloakEnabledBaseTestIT.java]: Have added comments to the base test to explain this now

**Pull request merged**